### ISSUE TITLE
Display "social currency" metrics about patterns

### ISSUE THREAD

**Issue by ryelle** [WordPress/pattern-directory#100]
The design includes views and a favorite count on each pattern. 

- Get the view count (or it a "times copied" count), if we can
- Add that into the CPT's API response for each pattern
- Display it in the grid

![pattern](https://user-images.githubusercontent.com/541093/117689979-51bde480-b188-11eb-804e-3ad178958520.png)

**Comment by coreymckrill** [comment]
So maybe the "times copied" count is a postmeta value on the pattern post? Would that only count uses of the "Copy pattern" button on the wp.org pattern directory, or could we bump the counter when someone picks the pattern via the API from within their site?

**Comment by shaunandrews** [comment]
I think it would be good to record values for:
- Views: How many times has a pattern detail page been viewed. It might be nice to break this down based on _where_ the view was registered; In a query loop, a single pattern, and/or from within the editor.
- Copies: How many times a pattern's copy button has been clicked. It might be nice to break this down based on _where_ the copy button was clicked; In a query loop or on the single screen for the pattern.
- Insertions: How many times a pattern was inserted from within the editor.
- Favorite: How many time's a pattern's favorite button has been clicked. It might be nice to break this down based on _where_ the buttons as clicked; In a query loop, on a single screen, or within the editor.

**Comment by ryelle** [comment]
I think we'll get pushback if we try to send back metrics from a user's site (privacy concerns).

Do we need to worry about performance with the idea of incrementing a counter with every copy?

Could we use Jetpack/tracks to offload handling the metrics, or does that open a new can of privacy worms?

**Comment by StevenDufresne** [comment]
Can we start with more simple metrics and expand? When I look at something like dribbble.com, they count favorites and views as the social indicators. If we took the same approach we would only need to track things on wordpress.org. It may help scope this down for the initial launch and if those values aren't good enough, we could ask users what they think is worth tracking.

**Comment by shaunandrews** [comment]
Sure. The designs already show the views and favorites as the primary metrics. I was more thinking of what could be helpful in the future as we iterate on the directory and pattern discovery/exploration.

**Comment by ryelle** [comment]
I added a fake favorite count in #164, but I'd like some help getting the real value. I started with this query:

```
$count = $wpdb->get_var( $wpdb->prepare(
	"SELECT COUNT(*)
		FROM {$wpdb->usermeta}
		WHERE meta_key=%s
		AND meta_value=%d",
	META_KEY,
	$post->ID
) );
```

Which works locally, but I don't know a) if we can query the usermeta table like that on multisite and b) if this is a terrible idea on the large user set in wp.org. This would run 18 times per page. So I imagine this needs some kind of caching - maybe each pattern should save its favorite count and update when it's favorited? That could get out of sync, but maybe this query can be run on favorite to get the count after a favorite is saved, and sync that into post meta?

Thoughts, @coreymckrill @iandunn @tellyworth ?

(I haven't touched on the views count idea, but I'd like some input on that too, if anyone has thoughts)
